clients/src/main/java/org/apache/kafka/clients/KafkaClient.java

```diff
@@ -65,6 +65,13 @@
     public boolean connectionFailed(Node node);
 
     /**
+     * Check if the node with the given id is connected
+     * @param node the node
+     * @return true if the node is connected
+     */
+    public boolean isConnected(Node node);
```

Comment:
Maybe we don't need these methods anymore since we got rid of the retry mechanism?